desktop/core/src/desktop/js/apps/newimporter/ImporterFilePreview/SourceConfiguration/SourceConfiguration.tsx

```diff
@@ -0,0 +1,70 @@
+// Licensed to Cloudera, Inc. under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  Cloudera, Inc. licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React from 'react';
+import Select from 'cuix/dist/components/Select/Select';
+import ConfigureIcon from '@cloudera/cuix-core/icons/react/ConfigureIcon';
+import { i18nReact } from '../../../../utils/i18nReact';
+import { sourceConfigs } from '../../constants';
+import { FileFormatResponse } from '../../types';
+
+import './SourceConfiguration.scss';
+
+interface SourceConfigurationProps {
+  fileFormat?: FileFormatResponse;
+  setFileFormat: (format: FileFormatResponse) => void;
+}
+const SourceConfiguration = ({
+  fileFormat,
+  setFileFormat
+}: SourceConfigurationProps): JSX.Element => {
+  const { t } = i18nReact.useTranslation();
+
+  const onChange = (value: string | number | boolean, name: keyof FileFormatResponse) => {
```

Comment:
we can use useCallback to avoid create new instances of onchange on every render.

Reply:
you can.... and then you could add `setFileFormat` to the dependency array for the useCallback, and making sure that function is not reinstantiated in the parent etc... but chances are you are just adding premature optimisations you won't notice and complexity you don't need.